src/client/visualizations/base-visualization/base-visualization.tsx

```diff
@@ -79,33 +82,45 @@ export class BaseVisualization<S extends BaseVisualizationState> extends React.C
 
   componentWillReceiveProps(nextProps: VisualizationProps) {
     if (this.shouldFetchData(nextProps) && this.visualisationNotResized(nextProps)) {
-      this.fetchData(nextProps.essence, nextProps.timekeeper);
+      this.loadData(nextProps.essence, nextProps.timekeeper, !this.props.essence.equals(nextProps.essence));
```

Comment:
I would extract this.props.essence and nextProps.essence to local variables. Or even better, the expression !this.props.essence.equals(nextProps.essence) should be assigned to local variable showSpinner, right now the meaning of latest method parameter is not clear for reader.